### PR TITLE
Quantify randomness of shuffle in squirrel

### PR DESCRIPTION
**Pull request by winfried-ripken**
# Description

Introduce a function to measure the randomness of a shuffle operation in the squirrel pipeline by implementing a simple example driver, random sampling and comparing the distances of sampled trajectories.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [x] Lint and unit tests pass locally with my changes
- [x] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

**Comment by winfried-ripken** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by winfried-ripken** [comment]
recheck

File: test/test_driver/test_quantify_randomness.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+@pytest.mark.parametrize("shards", [5, 10])
+@pytest.mark.parametrize("shard_size", [5, 10])
+def test_async_map_on_dataframe(shards: int, shard_size: int) -> None:
+    """Test that shuffled sequences produce more randomness."""
+    ktau_deterministic = quantify_randomness(shards, shard_size, 1, 1)
+    ktau_quasi_random = quantify_randomness(shards, shard_size, shard_size // 2, shard_size // 2)
+    ktau_fully_random = quantify_randomness(shards, shard_size, shard_size, shard_size)
+
+    assert ktau_deterministic > ktau_quasi_random > ktau_fully_random
```
Comment (AlirezaSohofi) [inline test/test_driver/test_quantify_randomness.py:14]: please add assertion for upper and lower bounds

File: test/test_driver/test_quantify_randomness.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import pytest
+
+from squirrel.driver.quantify_randomness import quantify_randomness
+
+
+@pytest.mark.parametrize("shards", [5, 10])
+@pytest.mark.parametrize("shard_size", [5, 10])
+def test_async_map_on_dataframe(shards: int, shard_size: int) -> None:
+    """Test that shuffled sequences produce more randomness."""
+    ktau_deterministic = quantify_randomness(shards, shard_size, 1, 1)
```
Comment (AlirezaSohofi) [inline test/test_driver/test_quantify_randomness.py:10]: maybe the default `250` samples is not really needed for test. wdyt?

**Review by AlirezaSohofi — CHANGES_REQUESTED** [review]
Well done, the implementation looks good. I think it does not really belong to the `driver` module, we can add a new module `benchmark`. 

Please also add this to the documentation.

File: docs/advanced/randomness.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Measure randomness of map driver
+==================================
+The :py:class:`~squirrel.driver.MapDriver` supports loading data from multiple shards.
+A difficult choice is often to select an appropriate shard size when a dataset is created.
+The choice of the shard size in combination with the size of the squirrel shuffel buffer determines
+if the iid assumption holds approximately while we sample the data.
```
Comment (AlirezaSohofi) [inline docs/advanced/randomness.rst:6]:
```suggestion
if the iid assumption holds approximately while we sample the data.
On the other hand, bigger shard size improves the loading performance
 substantially. Therefore we should aim for the biggest shard size that 
 does not hurt the approximate iid assumption. 
```

**Review by AlirezaSohofi — APPROVED** [review]
LGTM :+1:

**Comment by AlirezaSohofi** [comment]
@winfried-ripken Please don't forget to re-base and bump the version

**Pull request merged**